version: "3.11"

services:
  web:
    build:
      context: .
      dockerfile: ./Dockerfile
    command: uvicorn src.main:app --host 0.0.0.0 --port 80
    ports:
      - "80:80"
    environment:
      - PYTHONPATH=.
      - DATABASE_URL=postgres://labs:analytics@postgres:5432/lab-analytics?sslmode=disable
      - REDIS_URL=redis://redis_stack:6379/0
    depends_on:
      - redis_stack
    networks:
      - app-network

  postgres:
    image: postgres:alpine
    environment:
      POSTGRES_DB: lab-analytics
      POSTGRES_PASSWORD: analytics
      POSTGRES_USER: labs
    ports:
      - 5432:5432
    restart: on-failure:3
    volumes:
      - postgres:/var/lib/postgresql/data
    networks:
      - app-network

  pgweb:
    image: sosedoff/pgweb:0.15.0
    depends_on:
      - postgres
    environment:
      PGWEB_DATABASE_URL: postgres://labs:analytics@postgres:5432/lab-analytics?sslmode=disable
    ports:
      - 8002:8081
    restart: on-failure:3
    networks:
      - app-network

  redis_stack:
    image: redis/redis-stack:6.2.6-v6
    ports:
      - 6379:6379
      - 8001:8001
    volumes:
      - redis:/data
    networks:
      - app-network

networks:
  app-network:
    driver: bridge

volumes:
  redis:
  postgres:
